examples/twittermap/web/public/javascripts/common/services.js

```diff
@@ -434,6 +435,39 @@ angular.module('cloudberry.common', ['cloudberry.mapresultcache'])
             // unrecognized map type
             break;
         }
+      },
+
+      querypin: function(pinid){
+
+          var setpinfilter = [{
+              field: "id",
+              relation: "in",
+              values: pinid
+          }];
+
+          //sql
+          var pinDBresult = (JSON.stringify({
+              dataset:"twitter.ds_tweet",
+              filter: setpinfilter,
+              select:{
+                  order: ["-create_at"],
+                  limit: defaultSamplingSize,
+                  offset: 0,
+                  field: ["text","user.id","create_at","user.name","user.profile_image_url"]
+              },
+              transform:{
+                  wrap:{
+                      id:"pinDBtweet",
+                      category:"pinDBtweet"
+                  }
+              }
+          }));
+
+
+          //changes the id's type from string to int64.
+          pinDBresult = pinDBresult.replace(/"(\d+)"/,'[$1]');
```

Comment:
What does this do? I mean, why is this needed?

Reply:
As the comment said, the id is string and will be truncated and lost some digits. That's a critical bug I encountered when I do this task. Then we just remove the "" to let cloudberry accept. In Twitter's Database, it has two id's field and type is int and string to avoid this problem.